From 168a3ab126a77006065076867997d2f16f72d35f Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 12 Dec 2021 00:48:20 +0000
Subject: [PATCH 2/6] gnu: Add msgpack v4

* gnu/packages/golang.scm: (go-github-com-vmihailenco-msgpack-v4): New variable.
---
 gnu/packages/golang.scm | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index e484abd8e6..616a86cb99 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -8945,3 +8945,31 @@ (define-public go-github-com-vmihailenco-tagparser
     (synopsis "Tag parser for Golang")
     (description "Simple Golang implementation of tag parser")
     (license license:bsd-2)))
+
+(define-public go-github-com-vmihailenco-msgpack-v4
+  (package
+    (name "go-github-com-vmihailenco-msgpack-v4")
+    (version "4.3.12")
+    (source
+      (origin
+        (method git-fetch)
+        (uri (git-reference
+               (url "https://github.com/vmihailenco/msgpack")
+               (commit (string-append "v" version))))
+        (file-name (git-file-name name version))
+        (sha256
+          (base32 "0aiavk7b5fn050bbc0naldk2bsl60f8wil5i6a1cfp3lxxnvmvng"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/vmihailenco/msgpack/v4"))
+    (native-inputs
+     `(("go-gopkg-in-check-v1" ,go-gopkg-in-check-v1)))
+    (propagated-inputs
+     `(("go-github-com-vmihailenco-tagparser"
+        ,go-github-com-vmihailenco-tagparser)))
+    (home-page "https://github.com/vmihailenco/msgpack")
+    (synopsis "MessagePack encoding for Golang")
+    (description
+     "This package provides implementation of MessagePack encoding for Go
+programming language")
+    (license license:bsd-2)))
-- 
2.34.0

